utils/src/main/java/datadog/instrument/utils/ClassInfoCache.java

```diff
@@ -132,6 +133,47 @@ public T find(CharSequence className, int classLoaderKeyId) {
     }
   }
 
+  /**
+   * Finds information for the given class-name, filtered by class-loader key.
+   *
+   * @param className the class-name
+   * @param classLoaderKeyFilter the filter for the class-loader key
+   * @return information shared under the class-name and filtered class-loader
+   * @see ClassLoaderIndex#getClassLoaderKeyId(ClassLoader)
+   */
+  @SuppressWarnings("unchecked")
+  public T find(CharSequence className, IntPredicate classLoaderKeyFilter) {
```

Comment:
Note this is structurally very similar to the other `find` method, but I don't want to deduplicate the code for performance reasons. The other method uses a fast primitive operator inside the loop, whereas this adds a method call.

Reply:
Might be worth leaving a text comment here, for anyone opening the source, and not having the PR context :)